lang/python/python-s3transfer/Makefile

```diff
@@ -0,0 +1,57 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=s3transfer
+PKG_VERSION:=0.2.0
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://files.pythonhosted.org/packages/source/s/s3transfer
+PKG_HASH:=f23d5cb7d862b104401d9021fc82e5fa0e0cf57b7660a1331425aab0c691d021
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(BUILD_VARIANT)-s3transfer-$(PKG_VERSION)
+
+PKG_MAINTAINER:=Daniel Danzberger <daniel@dd-wrt.com>
+PKG_LICENSE:=MIT
+
+include $(INCLUDE_DIR)/package.mk
+include ../python-package.mk
+include ../python3-package.mk
+
+PKG_UNPACK:=$(HOST_TAR) -C $(PKG_BUILD_DIR) --strip-components=1 -xzf $(DL_DIR)/$(PKG_SOURCE)
+
+define Package/python-s3transfer/Default
+  SECTION:=lang
+  CATEGORY:=Languages
+  SUBMENU:=Python
+  TITLE:=s3transfer
+  URL:=https://github.com/boto/s3transfer
+endef
+
+define Package/python-s3transfer
+$(call Package/python-s3transfer/Default)
+  DEPENDS:=+python
```

Comment:
Comparing with [`install_requires` in setup.py](https://github.com/boto/s3transfer/blob/0.2.0/setup.py#L13-L21), this appears to be missing python-botocore (for both Python 2 and 3) and python-futures (for Python 2 only) as dependencies.

Reply:
I double checked the depenencies against the setup.py file for each pacakge.
Most of them seem correct and I updated the depends in their Makefile. (for s3transfer and rsa)

But botocore however has depenencies like pyasn1 which are not specified in setup.py.
Same goes for awscli with dateutils.

So I kept them as depeneds and successfully run tested them for each python variant.

Reply:
So you have mentioned pyasn1 for botocore, and dateutil for awscli. What about the other dependencies I have mentioned?

I am fairly certain simplejson is not necessary for botocore, because we are not using Python 2.6 and [it's actually explained in botocore's setup.py](https://github.com/boto/botocore/blob/1.12.66/setup.py#L35-L40)).

I appreciate that you have revised the dependencies for rsa and s3transfer. I would appreciate it if you can take a serious look at awscli and botocore as well.

As a reference, we can look at how the equivalent Debian packages have set up their dependencies: [awscli](https://packages.debian.org/buster/awscli), [botocore](https://packages.debian.org/buster/python3-botocore).

Reply:
You are right about simplejson, it works fine without.
I also used the debian packages as reference for awscli and botocore and successfully run tested them again for each variant.

Reply:
Thanks for the changes :smile:

I see you added python-requests as a dependency for python-botocore... The reason why the Debian package has this dependency is because botocore uses a [vendored copy of requests](https://github.com/boto/botocore/tree/1.12.66/botocore/vendored/requests), which Debian [replaces with the requests package](https://salsa.debian.org/python-team/modules/python-botocore/blob/master/debian/patches/0001-Don-t-use-duplicated-modules.patch).

Since this is something I'd like to see happen with this package as well (I might get to it later), I think it would be ok to merge this as it is.

Reply:
Ok, let's merge it! :-)

Reply:
@BKPepe it's resolved now (both the comment and the issue) :joy:

Reply:
Merged.